### ISSUE TITLE
Investigating and Enabling Autograd Support

### ISSUE THREAD

**Issue by Bmete7** [Munich-Quantum-Software-Stack/MQSS-Pennylane-Adapter#31]
We need to first investigate, if `MQSSPennylaneAdapter` is able to run in Autograd mode and can use parameter-shift method: 
```
import pennylane as qml
from pennylane import numpy as np
from mqss.pennylane_adapter.device import MQSSPennylaneDevice

dev = MQSSPennylaneDevice(wires=2, token=MQSS_TOKEN, backends=MQSS_BACKENDS)

@qml.qnode(dev, interface="autograd", diff_method="parameter-shift", shots = 1024)
def quantum_function_autograd(x, y):
    qml.RZ(x, wires=0)
    qml.CNOT(wires=[0, 1])
    qml.RY(y, wires=1)
    qml.CNOT(wires=[1, 0])
    qml.RX(x, wires=1)
    return qml.expval(qml.PauliX(0) @ qml.PauliZ(1)) # Here, qubits 0 and 1 are being measured, in bases X and Z, respectively. 

params = np.array([0.1, 0.2], requires_grad=True)
results = qml.gradients.param_shift(quantum_function_autograd)(params[0], params[1])
```

To-Do's:
- Figuring out whether a scheme similar to described above works in latest stable version of MQSS Pennylane Adapter
- If it can't, implement the required changes on the device level.

**Comment by Bmete7** [comment]
FYI @annkay108 
Additional bug tracing and user reports shared:

> I just tried to run my QNN with the updated MQSS-Pennylane adapter. The good news is that the forward pass on the device now worked fine. 😊
> On the other side, unfortunately, I still get an error related to the backward pass, which apparently fails at a null_postprocessing check. See the full error trace below.
> As you already mentioned, this concerns the parameter-shift method.
>  

> Some more details:
> I am using the device
> @qml.qnode(

```
MQSSPennylaneDevice(wires=2, token=os.getenv("<LRZ_API_TOKEN>"), backends=”EQE1”),
interface="torch",
diff_method="parameter-shift"
)
```

> I verified, that after one forward pass, the output of my quantum model looks like this:

     ` [tensor(-0.9400, dtype=torch.float64, grad_fn=<ExecuteTapesBackward>), tensor(0.5600, dtype=torch.float64, grad_fn=<ExecuteTapesBackward>)].`

> Meaning that for gradient calculation in backpropagation, it will call Pennylanes ExecuteTapesBackward, which is correct (no gradient call assignment missing).
> The list of post-processing results that pennylane checks in the backward pass are apparently empty.
> Since it works with the pennylane simulator (qml.device("default.qubit")), there must be then an issue in the adapter about passing on the gradients?


> Do you have an idea, what is the problem here?
